test/indexer/test_hca_indexer.py

```diff
@@ -106,6 +105,79 @@ class BundleAndSize(NamedTuple):
                 finally:
                     self._delete_indices()
 
+    def test_bundle_delete_downgrade(self):
+        """
+        Delete an updated version of a bundle, and ensure that the index reverts to the previous bundle.
+        """
+        self._index_canned_bundle(self.old_bundle)
+        old_hits_by_id = self._assert_old_bundle()
+        self._index_canned_bundle(self.new_bundle)
+        self._assert_new_bundle(num_expected_old_contributions=4, old_hits_by_id=old_hits_by_id)
+        self._delete_bundle(self.new_bundle)
+        self._assert_old_bundle(ignore_deletes=True)
+
+    def test_multi_entity_contributing_bundles(self):
+        """
+        Delete a bundle which shares entities with another bundle and ensure shared entities
+        are not deleted. Only entity associated with deleted bundle should be marked as deleted.
+        """
+        bundle_fqid = ("8543d32f-4c01-48d5-a79f-1c5439659da3", "2018-03-29T143828.884167Z")
+        patched_bundle_fqid = ("9654e431-4c01-48d5-a79f-1c5439659da3", "2018-03-29T153828.884167Z")
+        manifest, metadata = self._load_canned_bundle(bundle_fqid)
+        patched_manifest, patched_metadata = self._patch_bundle(manifest, metadata)
+        self._index_bundle(bundle_fqid, manifest, metadata)
+        self._index_bundle(patched_bundle_fqid, patched_manifest, patched_metadata)
+
+        hits = self._get_es_results()
+        num_docs_by_index_before = self._num_docs_by_index(hits)
+
+        self._delete_bundle(bundle_fqid)
+
+        hits_del = self._get_es_results()
+        num_docs_by_index_after = self._num_docs_by_index(hits_del)
+
+        for entity_type, aggregate in num_docs_by_index_after.keys():
+            # Both bundles reference two files. The share one file and exclusively own another one.
+            # Deleting one of the bundles, removes the file owned exclusively by that bundle.
+            difference = 1 if entity_type == 'files' and aggregate else 0
+            self.assertEqual(num_docs_by_index_after[entity_type, aggregate],
+                             num_docs_by_index_before[entity_type, aggregate] - difference)
+        for document in hits_del:
+            if document['_id'].startswith('b2cd6fb7-6612-4ab9-a1c5-e475663437e3_8543d32f-4c01-48d5-a79f-1c5439659da3'):
+                self.assertTrue(document['_source']['bundle_deleted'])
+
+    def _patch_bundle(self, manifest, metadata):
+        new_file_uuid = str(uuid4())
+        manifest = copy.deepcopy(manifest)
+        file_name = '21935_7#154_2.fastq.gz'
+        for i in range(len(manifest)):
```

Comment:
```suggestion
        for file in manifest:
```

Reply:
This works because you're replacing an element of the manifest entry, not the entire entry.